### DISCUSSION TITLE
When running the hexo server, can the content be deployed to the github page at the same time?

### DISCUSSION THREAD

**Discussion by huangsijun17** [hexojs/hexo#5177] [Question / Answer] [answered]
When running the hexo server, can the content be deployed to the github page at the same time?

**Comment by lorezyra** [comment]
Why? That makes no sense. The purpose of running `hexo server` is to preview your post _before_ you deploy it. Once you are happy with it, you can run `hexo deploy`. 

If you are certain the post is ready to deploy, then you could combine the commands together: `hexo s && hexo deploy`...

**Reply by huangsijun17** [reply]
I have been running `hexo server` continuously for a long time.

**Reply by lorezyra** [reply, marked as answer]
Hexo's software architecture is designed for a SSG (server side generated) webpage. It's not meant to be run 24/7 as the server...

## Reference
- https://blog.richiebartlett.com/Work_%E4%BB%95%E4%BA%8B/Programming_%E9%96%8B%E7%99%BA/Web-design-patterns-SSR-SSG-SPA/

**Reply by lorezyra** [reply]
That said, nothing stops you from running multiple instances of hexo on different ports. Just be aware that you could trigger race conditions and corrupt/lose your files.
